public/app/features/transformers/calculateHeatmap/heatmap.ts

```diff
@@ -292,7 +293,14 @@ export function prepBucketFrames(frames: DataFrame[]): DataFrame[] {
   }));
 }
 
-export function calculateHeatmapFromData(frames: DataFrame[], options: HeatmapCalculationOptions): DataFrame {
+interface HeatmapCalculationOptionsWithTimeRange extends HeatmapCalculationOptions {
+  timeRange?: TimeRange;
```

Comment:
will probably change this to generic xRange in the future, when Heatmap "officially" supports non-time X :)